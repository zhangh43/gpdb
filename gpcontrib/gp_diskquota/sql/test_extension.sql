CREATE DATABASE dbx0 ;
CREATE DATABASE dbx1 ;
CREATE DATABASE dbx2 ;
CREATE DATABASE dbx3 ;
CREATE DATABASE dbx4 ;
CREATE DATABASE dbx5 ;
CREATE DATABASE dbx6 ;
CREATE DATABASE dbx7 ;
CREATE DATABASE dbx8 ;
CREATE DATABASE dbx9 ;
CREATE DATABASE dbx10 ;

show max_worker_processes;

\! sleep 4
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l

\c dbx0
CREATE EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l
CREATE SCHEMA SX;
CREATE TABLE SX.a(i int);
SELECT diskquota.set_schema_quota('SX', '1MB');
INSERT INTO SX.a values(generate_series(0, 100000000));
INSERT INTO SX.a values(generate_series(0, 10));
DROP TABLE SX.a;

\c dbx1
CREATE EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l
CREATE SCHEMA SX;
CREATE TABLE SX.a(i int);
SELECT diskquota.set_schema_quota('SX', '1MB');
INSERT INTO SX.a values(generate_series(0, 100000000));
INSERT INTO SX.a values(generate_series(0, 10));
DROP TABLE SX.a;

\c dbx2
CREATE EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l
CREATE SCHEMA SX;
CREATE TABLE SX.a(i int);
SELECT diskquota.set_schema_quota('SX', '1MB');
INSERT INTO SX.a values(generate_series(0, 100000000));
INSERT INTO SX.a values(generate_series(0, 10));
DROP TABLE SX.a;

\c dbx3
CREATE EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l
CREATE SCHEMA SX;
CREATE TABLE SX.a(i int);
SELECT diskquota.set_schema_quota('SX', '1MB');
INSERT INTO SX.a values(generate_series(0, 100000000));
INSERT INTO SX.a values(generate_series(0, 10));
DROP TABLE SX.a;

\c dbx4
CREATE EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l
CREATE SCHEMA SX;
CREATE TABLE SX.a(i int);
SELECT diskquota.set_schema_quota('SX', '1MB');
INSERT INTO SX.a values(generate_series(0, 100000000));
INSERT INTO SX.a values(generate_series(0, 10));
DROP TABLE SX.a;

\c dbx5
CREATE EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l
CREATE SCHEMA SX;
CREATE TABLE SX.a(i int);
SELECT diskquota.set_schema_quota('SX', '1MB');
INSERT INTO SX.a values(generate_series(0, 100000000));
INSERT INTO SX.a values(generate_series(0, 10));
DROP TABLE SX.a;

\c dbx6
CREATE EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l
CREATE SCHEMA SX;
CREATE TABLE SX.a(i int);
SELECT diskquota.set_schema_quota('SX', '1MB');
INSERT INTO SX.a values(generate_series(0, 100000000));
INSERT INTO SX.a values(generate_series(0, 10));
DROP TABLE SX.a;

\c dbx7
CREATE EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l
CREATE SCHEMA SX;
CREATE TABLE SX.a(i int);
SELECT diskquota.set_schema_quota('SX', '1MB');
INSERT INTO SX.a values(generate_series(0, 100000000));
INSERT INTO SX.a values(generate_series(0, 10));
DROP TABLE SX.a;

\c dbx8
CREATE EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l
CREATE SCHEMA SX;
CREATE TABLE SX.a(i int);
SELECT diskquota.set_schema_quota('SX', '1MB');
INSERT INTO SX.a values(generate_series(0, 100000000));
INSERT INTO SX.a values(generate_series(0, 10));
DROP TABLE SX.a;

\c dbx9
CREATE EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l
CREATE SCHEMA SX;
CREATE TABLE SX.a(i int);
SELECT diskquota.set_schema_quota('SX', '1MB');
INSERT INTO SX.a values(generate_series(0, 10000000));
INSERT INTO SX.a values(generate_series(0, 10));
DROP TABLE SX.a;

\c dbx10
CREATE EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l

\c dbx0
DROP EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l

\c dbx1
DROP EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l

\c dbx2
DROP EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l

\c dbx3
DROP EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l

\c dbx4
DROP EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l

\c dbx5
DROP EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l

\c dbx6
DROP EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l

\c dbx7
DROP EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l

\c dbx8
DROP EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l

\c dbx9
DROP EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l

\c dbx10
DROP EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l

\c postgres

DROP DATABASE dbx0 ;
DROP DATABASE dbx1 ;
DROP DATABASE dbx2 ;
DROP DATABASE dbx3 ;
DROP DATABASE dbx4 ;
DROP DATABASE dbx5 ;
DROP DATABASE dbx6 ;
DROP DATABASE dbx7 ;
DROP DATABASE dbx8 ;
DROP DATABASE dbx9 ;
DROP DATABASE dbx10 ;
