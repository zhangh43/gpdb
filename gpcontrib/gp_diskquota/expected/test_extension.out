CREATE DATABASE dbx0 ;
CREATE DATABASE dbx1 ;
CREATE DATABASE dbx2 ;
CREATE DATABASE dbx3 ;
CREATE DATABASE dbx4 ;
CREATE DATABASE dbx5 ;
CREATE DATABASE dbx6 ;
CREATE DATABASE dbx7 ;
CREATE DATABASE dbx8 ;
CREATE DATABASE dbx9 ;
CREATE DATABASE dbx10 ;
show max_worker_processes;
 max_worker_processes 
----------------------
 12
(1 row)

\! sleep 4
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l
2
\c dbx0
CREATE EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l
3
CREATE SCHEMA SX;
CREATE TABLE SX.a(i int);
SELECT diskquota.set_schema_quota('SX', '1MB');
 set_schema_quota 
------------------
 
(1 row)

INSERT INTO SX.a values(generate_series(0, 100000000));
ERROR:  schema's disk space quota exceeded with name:sx
INSERT INTO SX.a values(generate_series(0, 10));
ERROR:  schema's disk space quota exceeded with name:sx
DROP TABLE SX.a;
\c dbx1
CREATE EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l
4
CREATE SCHEMA SX;
CREATE TABLE SX.a(i int);
SELECT diskquota.set_schema_quota('SX', '1MB');
 set_schema_quota 
------------------
 
(1 row)

INSERT INTO SX.a values(generate_series(0, 100000000));
ERROR:  schema's disk space quota exceeded with name:sx
INSERT INTO SX.a values(generate_series(0, 10));
ERROR:  schema's disk space quota exceeded with name:sx
DROP TABLE SX.a;
\c dbx2
CREATE EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l
5
CREATE SCHEMA SX;
CREATE TABLE SX.a(i int);
SELECT diskquota.set_schema_quota('SX', '1MB');
 set_schema_quota 
------------------
 
(1 row)

INSERT INTO SX.a values(generate_series(0, 100000000));
ERROR:  schema's disk space quota exceeded with name:sx
INSERT INTO SX.a values(generate_series(0, 10));
ERROR:  schema's disk space quota exceeded with name:sx
DROP TABLE SX.a;
\c dbx3
CREATE EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l
6
CREATE SCHEMA SX;
CREATE TABLE SX.a(i int);
SELECT diskquota.set_schema_quota('SX', '1MB');
 set_schema_quota 
------------------
 
(1 row)

INSERT INTO SX.a values(generate_series(0, 100000000));
ERROR:  schema's disk space quota exceeded with name:sx
INSERT INTO SX.a values(generate_series(0, 10));
ERROR:  schema's disk space quota exceeded with name:sx
DROP TABLE SX.a;
\c dbx4
CREATE EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l
7
CREATE SCHEMA SX;
CREATE TABLE SX.a(i int);
SELECT diskquota.set_schema_quota('SX', '1MB');
 set_schema_quota 
------------------
 
(1 row)

INSERT INTO SX.a values(generate_series(0, 100000000));
ERROR:  schema's disk space quota exceeded with name:sx
INSERT INTO SX.a values(generate_series(0, 10));
ERROR:  schema's disk space quota exceeded with name:sx
DROP TABLE SX.a;
\c dbx5
CREATE EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l
8
CREATE SCHEMA SX;
CREATE TABLE SX.a(i int);
SELECT diskquota.set_schema_quota('SX', '1MB');
 set_schema_quota 
------------------
 
(1 row)

INSERT INTO SX.a values(generate_series(0, 100000000));
ERROR:  schema's disk space quota exceeded with name:sx
INSERT INTO SX.a values(generate_series(0, 10));
ERROR:  schema's disk space quota exceeded with name:sx
DROP TABLE SX.a;
\c dbx6
CREATE EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l
9
CREATE SCHEMA SX;
CREATE TABLE SX.a(i int);
SELECT diskquota.set_schema_quota('SX', '1MB');
 set_schema_quota 
------------------
 
(1 row)

INSERT INTO SX.a values(generate_series(0, 100000000));
ERROR:  schema's disk space quota exceeded with name:sx
INSERT INTO SX.a values(generate_series(0, 10));
ERROR:  schema's disk space quota exceeded with name:sx
DROP TABLE SX.a;
\c dbx7
CREATE EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l
10
CREATE SCHEMA SX;
CREATE TABLE SX.a(i int);
SELECT diskquota.set_schema_quota('SX', '1MB');
 set_schema_quota 
------------------
 
(1 row)

INSERT INTO SX.a values(generate_series(0, 100000000));
ERROR:  schema's disk space quota exceeded with name:sx
INSERT INTO SX.a values(generate_series(0, 10));
ERROR:  schema's disk space quota exceeded with name:sx
DROP TABLE SX.a;
\c dbx8
CREATE EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l
11
CREATE SCHEMA SX;
CREATE TABLE SX.a(i int);
SELECT diskquota.set_schema_quota('SX', '1MB');
 set_schema_quota 
------------------
 
(1 row)

INSERT INTO SX.a values(generate_series(0, 100000000));
ERROR:  schema's disk space quota exceeded with name:sx
INSERT INTO SX.a values(generate_series(0, 10));
ERROR:  schema's disk space quota exceeded with name:sx
DROP TABLE SX.a;
\c dbx9
CREATE EXTENSION diskquota;
ERROR:  too many database to monitor
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l
11
CREATE SCHEMA SX;
CREATE TABLE SX.a(i int);
SELECT diskquota.set_schema_quota('SX', '1MB');
ERROR:  schema "diskquota" does not exist
LINE 1: SELECT diskquota.set_schema_quota('SX', '1MB');
               ^
INSERT INTO SX.a values(generate_series(0, 10000000));
INSERT INTO SX.a values(generate_series(0, 10));
DROP TABLE SX.a;
\c dbx10
CREATE EXTENSION diskquota;
ERROR:  too many database to monitor
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l
11
\c dbx0
DROP EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l
10
\c dbx1
DROP EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l
9
\c dbx2
DROP EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l
8
\c dbx3
DROP EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l
7
\c dbx4
DROP EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l
6
\c dbx5
DROP EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l
5
\c dbx6
DROP EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l
4
\c dbx7
DROP EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l
3
\c dbx8
DROP EXTENSION diskquota;
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l
2
\c dbx9
DROP EXTENSION diskquota;
ERROR:  extension "diskquota" does not exist
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l
2
\c dbx10
DROP EXTENSION diskquota;
ERROR:  extension "diskquota" does not exist
\! sleep 2
\! ps -ef | grep postgres | grep "\[diskquota]" | grep -v grep | wc -l
2
\c postgres
DROP DATABASE dbx0 ;
DROP DATABASE dbx1 ;
DROP DATABASE dbx2 ;
DROP DATABASE dbx3 ;
DROP DATABASE dbx4 ;
DROP DATABASE dbx5 ;
DROP DATABASE dbx6 ;
DROP DATABASE dbx7 ;
DROP DATABASE dbx8 ;
DROP DATABASE dbx9 ;
DROP DATABASE dbx10 ;
